drivers/w5500/w5500.c

```diff
@@ -0,0 +1,468 @@
+
+static inline void send_addr(w5500_t *dev, uint16_t addr)
+{
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, (addr >> 8));
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, (addr & 0xff));
+#else
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, (addr & 0xff));
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, (addr >> 8));
+#endif
+}
+
+static uint8_t rreg(w5500_t *dev, uint16_t reg)
+{
+    uint16_t address = reg & 0x07ff;
+    uint8_t command = (uint8_t)((reg & 0xf800) >> 8u) | CMD_READ;
+
+    send_addr(dev, address);
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, command);
+    return spi_transfer_byte(dev->p.spi, dev->p.cs, false, 0u);
+}
+
+static void wreg(w5500_t *dev, uint16_t reg, uint8_t data)
+{
+    uint16_t address = reg & 0x07ff;
+    uint8_t command = (uint8_t)((reg & 0xf800) >> 8u) | CMD_WRITE;
+
+    send_addr(dev, address);
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, command);
+    spi_transfer_byte(dev->p.spi, dev->p.cs, false, data);
+}
+
+static uint16_t raddr(w5500_t *dev, uint16_t addr_high, uint16_t addr_low)
+{
+    uint16_t res = (rreg(dev, addr_high) << 8);
+
+    res |= rreg(dev, addr_low);
+    return res;
+}
+
+static void waddr(w5500_t *dev,
+                  uint16_t addr_high, uint16_t addr_low, uint16_t val)
+{
+    wreg(dev, addr_high, (uint8_t)(val >> 8));
+    wreg(dev, addr_low, (uint8_t)(val & 0xff));
+}
+
+static void rchunk(w5500_t *dev, uint16_t addr, uint8_t *data, size_t len)
+{
+    uint16_t address = addr & 0x07ff;
+    uint8_t command = (uint8_t)((addr & 0xf800) >> 8u) | CMD_READ;
+
+    send_addr(dev, address);
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, command);
+    spi_transfer_bytes(dev->p.spi, dev->p.cs, false, NULL, data, len);
+}
+
+static void wchunk(w5500_t *dev, uint16_t addr, uint8_t *data, size_t len)
+{
+    uint16_t address = addr & 0x07ff;
+    uint8_t command = (uint8_t)((addr & 0xf800) >> 8u) | CMD_WRITE;
+
+    send_addr(dev, address);
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, command);
+    spi_transfer_bytes(dev->p.spi, dev->p.cs, false, data, NULL, len);
+}
+
+static void extint(void *arg)
+{
+    w5500_t *dev = (w5500_t *)arg;
+
+    netdev_trigger_event_isr(&dev->netdev);
+    if (dev->p.polling_interval_ms > 0u) {
+        /* restart timer if we are polling */
+        (void)ztimer_set(ZTIMER_MSEC, &dev->timerInstance, dev->p.polling_interval_ms);
+    }
+}
+
+void w5500_setup(w5500_t *dev, const w5500_params_t *params, uint8_t index)
+{
+    assert(dev);
+    assert(params);
+
+    /* Initialize netdev structure. */
+    dev->netdev.driver = &netdev_driver_w5500;
+    dev->netdev.event_callback = NULL;
+    dev->netdev.context = dev;
+    /* Initialize the device descriptor. */
+    dev->p = *params;
+
+    dev->link_up = false;
+
+    /* Initialize the chip select pin. */
+    spi_init_cs(dev->p.spi, dev->p.cs);
+    if (dev->p.polling_interval_ms == 0u) {
```

Comment:
Ok. I changed w5500_params.h so that GPIO_UNDEF is set for W5500_PARAM_EVT (instead of copying the value from W5500_PARAM_CS) to make it clear that W5500_PARAM_EVT is invalid. I also modified the line you recommended.
Nevertheless I kept the existing code in init() and extint() where the value of polling_interval_ms is checked to decide if the ztimer should be started. I hope this is not confusing. 
When polling mode is desired: 
- A valid polling_interval_ms has to be set AND W5500_PARAM_EVT has to be GPIO_UNDEF. 
When interrupt mode is desired:
- polling_interval_ms has to be 0 AND W5500_PARAM_EVT has to be valid 

I have added an assert in w5500_setup() to ensure the parameters are properly set.